src/main/java/com/internet/shop/dao/jdbc/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,202 @@
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not get user from DB with login = " + login, e);
+        }
+        if (user != null) {
+            user.setRoles(getRolesOfUser(user.getId()));
+        }
+        return Optional.ofNullable(user);
+    }
+
+    @Override
+    public User create(User user) {
+        String query = "INSERT INTO users(name, login, password) VALUES(?, ?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, user.getName());
+            statement.setString(2, user.getLogin());
+            statement.setString(3, user.getPassword());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                user.setId(resultSet.getLong(1));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to create user " + user, e);
+        }
+        return defineRoles(user);
+    }
+
+    @Override
+    public Optional<User> getById(Long userId) {
+        User user = null;
+        String query = "SELECT * FROM users u "
+                + "INNER JOIN users_roles ur ON u.user_id = ur.user_id "
+                + "WHERE u.deleted = FALSE AND u.user_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, userId);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                user = getUserFromResultSet(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to create user with ID " + userId, e);
+        }
+        if (user != null) {
+            user.setRoles(getRolesOfUser(user.getId()));
+        }
+        return Optional.ofNullable(user);
+    }
+
+    @Override
+    public User update(User user) {
+        String query = "UPDATE users SET name = ?, login = ?, password = ? "
+                + "WHERE user_id = ? AND deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, user.getName());
+            statement.setString(2, user.getLogin());
+            statement.setString(3, user.getPassword());
+            statement.setLong(4, user.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to create user " + user, e);
+        }
+        deleteRoles(user.getId());
+        defineRoles(user);
+        user.setRoles(getRolesOfUser(user.getId()));
```

Comment:
this line is redundant, user in update already has roles, besides you've done in defineRoles() anyway

Reply:
Done.